### PR TITLE
Exploration page

### PR DESCRIPTION
**Pull request by sonalkr132**
- [x] give a project of rating 1-5 on both user-projects show page and project-show page
- [x] user-projects page will show average rating of project and project-show page will have rating of current user on that project
- [x] Write tests
- [x] make inspiration page into an exploration page with option to sort on basis of average ratings, number of followers, number of forks, most active in past 10 weeks (bases on no of commits and issue closed)
- [x] write test for sorting (needs help)
- [x] better image layout of project page (pending review)
- [x] write test for responsive images rendering on inspire page

**#Needs_Feedback ~~#In_Progress~~**

### CODE REVIEW

**Comment by sarupbanskota** [comment]
Notes from the meeting: 

* Add sort based on - activity, last updated.

**Comment by sarupbanskota** [comment]
@AkshataDM could you please also add a link to a production demo (on heroku, if need be) for me to quickly check on your progress?

**Comment by sonalkr132** [comment]
sorry, I can't. I have all my heroku apps occupied. I would need to delete either 'tagging' or 'ajax comments' deployment, if you want me to deploy this one. 
Rest of three are occupied with my personal projects.

**Comment by sarupbanskota** [comment]
You can delete tagging, I've reviewed it and will merge it when I next have time.

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -20,14 +20,42 @@ def new
...
+                                   AND comments.polycomment_type='project'
+                                   LEFT OUTER JOIN issues ON issues.project_id = projects.id
+                                   AND issues.status=1")
+                            .where("comments.created_at > ? OR issues.updated_at > ?", 10.days.ago, 10.days.ago)
+                            .group("projects.id")
+                            .order("count(comments.polycomment_id)+4*count(issues.project_id) desc")
+      else
+        @projects = Project.order('created_at DESC')
+      end
+    end
```
Comment (sonalkr132) [inline app/controllers/projects_controller.rb]: @rohitpaulk Do these queries look okay? They take less than 1ms but I am not sure how are they going to scale.

Also, Any tips about how can I avoid writing really lengthy test for this. As far as I can think, I will have to make multiple projects and then add multiple followers, comments, forks, issues and stars to each one of them. That must take atleast 200 lines.

**Comment by sonalkr132** [comment]
demo: http://ruby-glitterfollowers.rhcloud.com/

Go to inspire page after sign up and try out different options. I think we need to add something like github (crunching data...) cause it feels slow.

I have added first image as project image but it kinda screwing up structure of page. We can either set a fix height (ruins aspect ratio of image) or apply something like [this](http://jeremycowart.com/portfolio/featured/) (which would take time to implement).

Also, I need help with writing tests of sorting. I will be working on authorization until I get my questions answered here.

**Comment by sarupbanskota** [comment]
On Sat, May 23, 2015 at 3:04 PM, Aditya Prakash <notifications@github.com>
wrote:

> demo: http://ruby-glitterfollowers.rhcloud.com/
>
This looks good!

> Go to inspire page after sign up and try out different options. I think we
> need to add something like github (crunching data...) cause it feels slow.
>
Yes, you'll need to add a loader - I have more interesting ideas for
exploring here, but let's revisit them later.

> I have added first image as project image but it kinda screwing up
> structure of page. We can either set a fix height (ruins aspect ratio of
> image) or apply something like this
> <http://jeremycowart.com/portfolio/featured/> (which would take time to
> implement).
>
I think what currently exists is fine for now, but if you'd like to include
a quick improvement, take a look at Masonry: http://masonry.desandro.com.
When you're feeling up to a challenge, you can try a pure CSS solution to
emulate the Masonry effect using Flexbox or CSS3 columns, but you'll have
to introduce polyfills for browsers that don't play well with them. I used
to have an example here:
http://sarupbanskota.github.io/peanuts/watt-a-graphy/photos.html, although
this one was just an experiment so it doesn't include polyfills. Try it on
a latest FF, it uses CSS3 columns.


> Also, I need help with writing tests of sorting. I will be working on
> authorization until I get my questions answered here.
>
What sort of help do you require? cc @rohitpaulk.

**Comment by sarupbanskota** [comment]
Also, can you please make the inspire page the landing page, and make it visible without having to login.

**Comment by sonalkr132** [comment]
>make it visible without having to login

I am going to cover that part in authorization. We need to discuss what parts need to be visible without login. Check #300 to get idea of how I think things should be. If you have time, you can go through the tests I have added there. There are a quite a few things which we will need to revisit after git server integration.

> What sort of help do you require?

I have already mentioned him asking for help. He must not have time these days. 
I just think if I right test on my own it will be really lenghty. I guess I will just write whatever I know.

**Comment by sonalkr132** [comment]
This is now complete. However I am unable to deploy at the moment. Not many changes are made since my last deployment anyway. I added resize_to_fill for project images on inspire page. here is a screen shot:
screenshot from 2015-05-30 23 35 50

**Comment by sarupbanskota** [comment]
Please let me know when you have a deployment anyway, merging now to avoid blockers.

**Comment by sonalkr132** [comment]
Yeah sure. I will deploy #300. Which should be complete by tomorrow morning.

**Comment by sarupbanskota** [comment]
Updates?

On Saturday, May 30, 2015, Aditya Prakash <notifications@github.com> wrote:

> Yeah sure. I will deploy #300
> <https://github.com/glittergallery/GlitterGallery/pull/300>. Which should
> be complete by tomorrow morning.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/glittergallery/GlitterGallery/pull/298#issuecomment-107073957>
> .
>


-- 
Sarup

**Pull request merged**